### PR TITLE
Utility functions for `PMaybe` and `PMaybeData`

### PR DESCRIPTION
**Pull request by SeungheonOh**
A one of new PRs that will be made for Liqwid-Plutarch-Extra upstreaming.

### CODE REVIEW

File: plutarch-extra/Plutarch/Extra/Maybe.hs [hunk 1]
```diff
@@ -0,0 +1,204 @@
+{-# LANGUAGE PolyKinds #-}
```
Comment (L-as) [inline plutarch-extra/Plutarch/Extra/Maybe.hs:1]: ? You can remove all three of these.

File: plutarch-extra/Plutarch/Extra/Maybe.hs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+import Plutarch.String (PString)
+import Plutarch.TermCont (TermCont, tcont)
+import Plutarch.Trace (ptraceError)
+
+--------------------------------------------------------------------------------
+-- Utility functions for working with 'PMaybe'.
+
+-- | Extracts the element out of a 'PJust' and throws an error if its argument is 'PNothing'.
+pfromJust ::
+    forall (a :: S -> Type) (s :: S).
```
Comment (L-as) [inline plutarch-extra/Plutarch/Extra/Maybe.hs:60]: PType

**Review by L-as — CHANGES_REQUESTED** [review]

**Comment by L-as** [comment]
I can't merge this because the branch is out-of-date. I can't make it up-to-date because I can't modify the code in the branch.

**Comment by peter-mlabs** [comment]
Updated the branch

**Comment by SeungheonOh** [comment]
Thanks Peter, I'll do the same for `PMap` PR. For future PRs, do you want me to make branch on Plutonomicon repository? I doubt I have right permission for that.

File: plutarch-extra/Plutarch/Extra/Maybe.hs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+) where
+
+import Plutarch.Api.V1.Maybe (PMaybeData (PDJust, PDNothing))
+import Plutarch.Prelude
+
+--------------------------------------------------------------------------------
+-- Utility functions for working with 'PMaybe'.
+
+-- | Extracts the element out of a 'PJust' and throws an error if its argument is 'PNothing'.
+pfromJust ::
```
Comment (L-as) [inline plutarch-extra/Plutarch/Extra/Maybe.hs:36]: All partial functions should have a `HasCallStack` constraint.

**Review by L-as — CHANGES_REQUESTED** [review]

**Comment by L-as** [comment]
Please do it in another PR.

**Pull request merged**